ownCloud/Settings/LogFilesViewController.swift

```diff
@@ -0,0 +1,222 @@
+		self.title = "Log Files".localized
+	}
+
+	override func viewWillAppear(_ animated: Bool) {
+		super.viewWillAppear(animated)
+		self.populateLogFileList()
+
+		let removeAllButtonItem = UIBarButtonItem(title: "Delete all".localized, style: .done, target: self, action: #selector(removeAllLogs))
+		let flexibleSpaceButtonItem = UIBarButtonItem(barButtonSystemItem: .flexibleSpace, target: nil, action: nil)
+
+		self.toolbarItems = [flexibleSpaceButtonItem, removeAllButtonItem, flexibleSpaceButtonItem]
+		self.navigationController?.setToolbarHidden(false, animated: false)
+
+		NotificationCenter.default.addObserver(self, selector: #selector(handleLogRotationNotification), name:NSNotification.Name.OCLocalLogRotationNotificationName , object: nil)
+	}
+
+	override func viewWillDisappear(_ animated: Bool) {
+		super.viewWillDisappear(animated)
+		self.navigationController?.setToolbarHidden(true, animated: false)
+		NotificationCenter.default.removeObserver(self)
+	}
+
+	deinit {
+		Theme.shared.unregister(client: self)
+	}
+
+	// MARK: - Theme support
+
+	func applyThemeCollection(theme: Theme, collection: ThemeCollection, event: ThemeEvent) {
+		self.tableView.applyThemeCollection(collection)
+
+		if event == .update {
+			self.tableView.reloadData()
+		}
+	}
+
+	// MARK: - Table view data source / delegate
+
+	override func numberOfSections(in tableView: UITableView) -> Int {
+		return 1
+	}
+
+	override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+		return logRecords.count
+	}
+
+	override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+		let cell = tableView.dequeueReusableCell(withIdentifier: LogFileTableViewCell.identifier, for: indexPath) as? LogFileTableViewCell
+		let logEntry = self.logRecords[indexPath.row]
+		cell?.textLabel?.text = logEntry.truncatedName()
+		if let date = logEntry.creationDate {
+			cell?.detailTextLabel?.text = "\(self.dateFormatter.string(from: date)), \(self.byteCounterFormatter.string(fromByteCount: logEntry.size))"
+		}
+
+		cell?.shareAction = { [weak self] (cell) in
+			if let indexPath = self?.tableView.indexPath(for: cell) {
+				self?.shareLogRecord(at: indexPath)
+			}
+		}
+
+		return cell!
+	}
+
+	override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+		tableView.deselectRow(at: indexPath, animated: true)
+	}
+
+	override func tableView(_ tableView: UITableView, editActionsForRowAt indexPath: IndexPath) -> [UITableViewRowAction]? {
+
+		return [
+			UITableViewRowAction(style: .destructive, title: "Delete".localized, handler: { [weak self] (_, indexPath) in
+				self?.removeLogRecord(at: indexPath)
+			})
+		]
+	}
+
+	// MARK: - Private Helpers
+
+	private func populateLogFileList() {
+		guard let logFileWriter = OCLogger.shared.writer(withIdentifier: .writerFile) as? OCLogFileWriter else { return }
+		self.logRecords = logFileWriter.logRecords()
```

Comment:
Maybe sort the result in reverse order so the latest log file is always on top. (I noticed you sort it from oldest to latest in the SDK as part of the log rotation routine, so the reverse sorting should likely happen in the app in order not to impact that)

Reply:
@felix-schwarz fixed